app.js

```diff
@@ -1,77 +1,164 @@
-  return Math.floor(Math.floor(Math.random() * img.length));
+function randomNum(){
+  return Math.floor(Math.random() * productNames.length);
 };
 
-/*function displayPics() {
+function displayFinalResults() {
+  console.log('displayFinalResults');
+  for (var i = 0; i < allProducts.length; i++) {
+    var productClicksEl = document.createElement('li');
+    console.dir(productClicksEl);
+    clicks.appendChild(productClicksEl);
+    productClicksEl.textContent = allProducts[i].name + ': clicks ' + allProducts[i].clicks + ', views ' + allProducts[i].views;
+  }
+}
+
+function generateImg() {
   var leftIndex = randomNum();
   var centerIndex = randomNum();
   var rightIndex = randomNum();
-}*/
 
-var allProducts = [];
+  console.log('Left Index: ' + leftIndex);
+  console.log('Center Index: ' + centerIndex);
+  console.log('Right Index: ' + rightIndex);
+//left image generator
+  while(previousImgs.includes(leftIndex)){
+    var leftIndex = randomNum ();
+    console.log('New left: ' + leftIndex);
+  }
 
-var leftIndex = randNum();
-var leftImg = allProducts[leftIndex];
-previousImgs.push(leftIndex);
+  while(centerIndex === leftIndex || previousImgs.includes(centerIndex)){
+    var centerIndex = randomNum();
+    console.log('New Center:' + centerIndex);
+  }
 
-var centerIndex = randNum();
-while (centerIndex === leftIndex) {
-  centerIndex = randNum();
-}
-var centerImg = allProducts[centerIndex];
-previousImgs.push(centerIndex);
+  while(rightIndex === leftIndex || rightIndex === centerIndex || previousImgs.includes(rightIndex)) {
+    var rightIndex = randomNum();
+    console.log('New Right: ' + rightIndex);
+  }
+
+  previousImgs = [leftIndex, centerIndex, rightIndex];
+
+  console.log(previousImgs);
+
+  var leftImgElement = document.createElement('img');
+  leftImgElement.setAttribute('src', allProducts[previousImgs[0]].path);
+  leftImgElement.setAttribute('alt',allProducts[previousImgs[0]].name);
+  imgContainer.appendChild(leftImgElement);
+
+  var centerImgElement = document.createElement('img');
+  centerImgElement.setAttribute('src', allProducts[previousImgs[1]].path);
+  centerImgElement.setAttribute('id', allProducts[previousImgs[1]].name);
+  imgContainer.appendChild(centerImgElement);
 
-var rightIndex = randNum();
-while(rightIndex === leftIndex || rightIndex === centerIndex){ var rightIndex = randNum();
+  var rightImgElement = document.createElement('img');
+  rightImgElement.setAttribute('src', allProducts[previousImgs[2]].path);
+  rightImgElement.setAttribute('id', allProducts[previousImgs[2]].name);
+  imgContainer.appendChild(rightImgElement);
 }
+//returns previousImgs:
+
+createProducts();
+generateImg();
+
+//on-click Listener
+imgContainer.addEventListener('click', handleImagesOnClick);
+//event listener for one box with all 3 imgNames
+function handleImagesOnClick(event) {
+  console.log(event);
+  for (var j = 0; j < allProducts.length; j++) {
+    if (event.target.alt === allProducts[j].name) {
```

Comment:
Great! 

After you find the one that matches, is there any point in continuing to iterate through the rest of the products? Can you think of a way to break out of the loop at that point?